src/additional_zones.rs

```diff
@@ -0,0 +1,216 @@
+use geo_types::{Coordinate, MultiPolygon, Point, Rect};
+use osmpbfreader::{OsmObj, OsmPbfReader};
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::path::Path;
+use crate::zone::{Zone, ZoneIndex, ZoneType};
+use crate::zone_tree::ZonesTree;
+use geos::from_geo::TryInto;
+use geos::GGeom;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+
+struct ZoneWithGeos<'a> {
+    zone: &'a Zone,
+    geos: GGeom,
+}
+
+unsafe impl<'a> Send for ZoneWithGeos<'a> {}
+unsafe impl<'a> Sync for ZoneWithGeos<'a> {}
+
+impl<'a> ZoneWithGeos<'a> {
+    fn new(zone: &'a Zone) -> ZoneWithGeos<'a> {
+        ZoneWithGeos {
+            zone,
+            geos: zone.boundary.as_ref().unwrap().try_into().expect("failed to convert to geos"),
+        }
+    }
+}
+
+pub fn compute_additional_cities(zones: &mut Vec<Zone>, pbf_path: &str) {
+    let place_zones = read_places(pbf_path);
+    info!(
+        "there are {} places, we'll try to make boundaries for them",
+        place_zones.len()
+    );
+    let zones_rtree: ZonesTree = zones.iter().filter(|z| z.is_admin()).collect();
```

Comment:
this `ZonesTree` seems to be equivalent to another `ZonesTree` existing in hierarchy_builder.rs, except `gst` crate was recently replaced with rstar. 
So we should be able to build it only once.